packages/shared/src/components/modals/SquadPostModal.tsx

```diff
@@ -0,0 +1,226 @@
+import { Origin } from '../../lib/analytics';
+import usePostContent from '../../hooks/usePostContent';
+import usePostById from '../../hooks/usePostById';
+import usePostNavigationPosition from '../../hooks/usePostNavigationPosition';
+import { ModalSize } from './common/types';
+import FixedPostNavigation from '../post/FixedPostNavigation';
+import SourceButton from '../cards/SourceButton';
+import { Separator } from '../cards/common';
+import { UserShortInfo } from '../profile/UserShortInfo';
+import PostSummary from '../cards/PostSummary';
+import { LazyImage } from '../LazyImage';
+import { Source } from '../../graphql/sources';
+import { ProfileImageSize } from '../ProfilePicture';
+import { ReadArticleButton } from '../cards/ReadArticleButton';
+import ArrowIcon from '../icons/Arrow';
+
+const SharePostModal = dynamic(
+  () => import(/* webpackChunkName: "shareModal" */ './ShareModal'),
+);
+
+interface PostModalProps
+  extends ModalProps,
+    Pick<PostNavigationProps, 'onPreviousPost' | 'onNextPost'> {
+  id: string;
+  isFetchingNextPage?: boolean;
+}
+
+interface SourceInfoProps {
+  source: Source;
+  date?: string;
+  size?: ProfileImageSize;
+  typo?: string;
+}
+
+const SourceInfo = ({
+  date,
+  source,
+  size = 'medium',
+  typo = 'typo-callout',
+}: SourceInfoProps) => {
+  return (
+    <span
+      className={classNames(
+        'flex flex-row items-center text-theme-label-tertiary',
+        typo,
+      )}
+    >
+      <SourceButton source={source} size={size} />
+      <h3 className="ml-3">{source.handle}</h3>
+      <Separator />
+      <time dateTime={date}>{date}</time>
+    </span>
+  );
+};
+
+export default function PostModal({
+  id,
+  className,
+  isFetchingNextPage,
+  onRequestClose,
+  onPreviousPost,
+  onNextPost,
+  ...props
+}: PostModalProps): ReactElement {
+  const hasNavigation = !!onPreviousPost || !!onNextPost;
+  const [height, setHeight] = useState<number>(null);
+  const [shoudShowSumamry, setShouldShowSummary] = useState(true);
+  const { post, isLoading } = usePostById({ id, isFetchingNextPage });
+  const position = usePostNavigationPosition({
+    isLoading,
+    isDisplayed: props.isOpen,
+  });
+  const { onCloseShare, onReadArticle, onShare, onToggleBookmark, sharePost } =
+    usePostContent({
+      origin: Origin.ArticleModal,
+      post,
+    });
+
+  if (!post) return <></>;
+
+  const navigationProps: PostNavigationProps = {
+    post,
+    onBookmark: onToggleBookmark,
+    onPreviousPost,
+    onNextPost,
+    onReadArticle,
+    onClose: onRequestClose,
+    onShare,
+    inlineActions: true,
+  };
+
+  const postLinkProps = {
+    href: post.sharedPost.permalink,
+    title: 'Go to article',
+    target: '_blank',
+    rel: 'noopener',
+    onClick: onReadArticle,
+    onMouseUp: (event: React.MouseEvent) =>
+      event.button === 1 && onReadArticle(),
+  };
+
+  const getHeight = () => {
+    if (height === null) return 'auto';
+
+    return shoudShowSumamry ? height : 0;
```

Comment:
shouldShowSummary

Reply:
I am plagued by typos. I think I need to get my eyes checked again 😆